st2common/tests/unit/test_misc_utils.py

```diff
@@ -94,3 +95,44 @@ def test_fast_deepcopy_success(self):
             result = fast_deepcopy(value)
             self.assertEqual(result, value)
             self.assertEqual(result, expected_value)
+
+    def test_sanitize_output(self):
+        # 1. pty is not used, \r\n shouldn't be replaced with \n
+        input_strs = [
+            'foo',
+            'foo\n',
+            'foo\r\n',
+            'foo\nbar\nbaz\n',
+            'foo\r\nbar\r\nbaz\r\n',
+        ]
+        expected = [
+            'foo',
+            'foo',
+            'foo',
+            'foo\nbar\nbaz',
+            'foo\r\nbar\r\nbaz',
+        ]
+
+        for input_str, expected_output in zip(input_strs, expected):
+            output = sanitize_output(input_str, uses_pty=False)
+            self.assertEqual(expected_output, output)
+
+        # 2. pty is used, \r\n should be replaced with \n
```

Comment:
Should this be split out into its own test function?

Reply:
I'm fine either way :)

Reply:
I decided to split it into two methods - 3524e05c2.

When to split the tests and when not to is not always totally clear - when a test needs to be isolated and potentially also requires on ``setUp`` and ``tearDown`` method to run then it's clear that such tests need to be separate methods, but in scenarios like that it's less clear / needed.

In general, I do agree with you though - I prefer separate methods since it provides more user-friendly test runner output (and unless setUp / tearDown add a huge overhead then there are really no drawbacks).